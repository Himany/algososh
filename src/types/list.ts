import { ElementStates } from "./element-states";

export type TListCircleObjects = {
  letter: number | string;
  state: ElementStates;
  headElement: number | string | null;
  tailElement: number | string | null;
}