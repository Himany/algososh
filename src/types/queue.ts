import { ElementStates } from "./element-states";

export type TQueueCircleObjects = {
  letter: string | null;
  state: ElementStates;
}