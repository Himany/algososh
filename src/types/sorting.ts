import { ElementStates } from "./element-states";

export type TColumnObjects = {
  index: number;
  state: ElementStates;
}