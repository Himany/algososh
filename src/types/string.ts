import { ElementStates } from "./element-states";

export type TCircleObjects = {
  letter: string;
  state: ElementStates;
}