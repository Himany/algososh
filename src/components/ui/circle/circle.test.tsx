import React from 'react';
import renderer from 'react-test-renderer';

import { Circle } from './circle';
import { ElementStates } from "../../../types/element-states";

describe('Circle test', () => {
  it('Circle without letter', () => {
    const tree = renderer
      .create(<Circle />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Circle with letter', () => {
    const tree = renderer
      .create(<Circle letter='T'/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Circle with letter in head', () => {
    const tree = renderer
      .create(<Circle head='T'/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with react element in head', () => {
    const tree = renderer
      .create(<Circle head={<h1>TEST</h1>}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with letter in tail', () => {
    const tree = renderer
      .create(<Circle tail='T'/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with react element in tail', () => {
    const tree = renderer
      .create(<Circle tail={<h1>TEST</h1>}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with index', () => {
    const tree = renderer
      .create(<Circle index={42}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with prop isSmall', () => {
    const tree = renderer
      .create(<Circle isSmall={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with state Default', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with state Changing', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
  it('Circle with state Modified', () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
});



