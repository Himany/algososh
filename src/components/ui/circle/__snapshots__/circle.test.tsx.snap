// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle test Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    42
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle test Circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with letter in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    T
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with letter in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    T
  </div>
</div>
`;

exports[`Circle test Circle with prop isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with react element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <h1>
      TEST
    </h1>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with react element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <h1>
      TEST
    </h1>
  </div>
</div>
`;

exports[`Circle test Circle with state Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with state Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle with state Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle test Circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
